File: ios/MobileOrders/MobileOrders/Models/AppManager.swift
//
//  AppManager.swift
//  MobileOrders
//
//  Created by Jacob Ilin on 4/25/20.
//  Copyright © 2020 Glowbom. All rights reserved.
//

import Foundation

class AppManager {
    static let instance = AppManager()

    private init() {
        self.orders = []
        self.total = 0
    }

    var orders: [Post]
    var total: Float
    
    func addOrder(order: Post) {
        orders.append(order)
        total += order.price
    }
}

File: android/MobileOrders/app/src/main/java/com/glowbom/mobileorders/model/AppManager.kt
/*
 * Created on 5/1/20 4:25 PM
 *
 * Copyright 2020 Glowbom, Inc.
 */

package com.glowbom.mobileorders.model

class AppManager (
    private val data: MutableList<Item>,
    private var total: Float
) {
    companion object {
        val instance = AppManager(mutableListOf(), 0f)
    }

    fun addOrder(order: Item) {
       data.add(order);
        order.let {
            order.price.let {
                total += order.price!!
            }
        }
    }

    fun clear() {
        total = 0f
        data.clear()
    };

    fun getTotal() : Float {
        return total;
    };
}